app/javascript/lib/Entities/AbstractImperialGame.js

```diff
@@ -8,14 +8,42 @@ export default class AbstractImperialGame extends Entity {
   }
 
   #rondel;
+  #availableFreeRondelSlotCount = 3;
+  #availablePaidRondelSlotCount = 3;
 
-  constructor(id) {
+  #nations;
+  #nationOrder;
+
+  constructor(id, nations, nationOrder) {
     super(id);
 
     this.#rondel = new Rondel();
+
+    this.#nations = nations;
+    this.#nationOrder = nationOrder;
   }
 
   get rondel() {
     return this.#rondel;
   }
+
+  get availableFreeRondelSlotCount() {
+    return this.#availableFreeRondelSlotCount;
+  }
+  get availablePaidRondelSlotCount() {
+    return this.#availablePaidRondelSlotCount;
+  }
+
+  nationIdToEntity(nationId) {
+    return this.#nations.get(nationId);
+  }
+
+  nationTurnAfter(nation) {
+    let nationIndex = this.#nationOrder.indexOf(nation);
+
+    nationIndex += 1;
+    nationIndex %= this.#nationOrder.length;
```

Comment:
What is this line doing?

Reply:
Not sure if js supports out of bound indices, just wrapping it around to make sure it's between `0 to length -1`